src/solvers/refinement/string_constraint_generator_valueof.cpp

```diff
@@ -451,7 +455,7 @@ exprt string_constraint_generatort::add_axioms_for_parse_int(
     for(unsigned j=1; j<size; j++)
     {
       mult_exprt radix_sum(sum, radix);
-      if(j>=9)
+      if(j>=max_string_length-1)
```

Comment:
Yes that's true, but I think there is a maximum radix (much smaller than the maximal int) so we need more than 2 characters in any case to cause an overflow